BLOG.md

```diff
@@ -0,0 +1,363 @@
+userapi   : ready to intercept (traffic-agent not yet installed)
+```
+
+How can you test your compose file against these services ?
+
+Typically, you have two options: either you continuously test everything locally, ensuring it closely resembles the real setup, or you maintain a separate test environment in a different namespace where you can update the pod with your test version.
+
+This is where Telepresence can be extremely beneficial for your setup, particularly because it offers integration with Docker Compose.
+
+## The Telepresence specification
+
+### Basic knowledge
+
+Telepresence supports a file format that it is possible to use to persist any type of configuration: The Intercept Specification.
+
+It has three sections :
+
+- `connection`: This one is optional, but it allows you to define which context the spec should be used with.
+- `workloads`: It defines the workloads to intercept, and how (header, ports, ...).
+- `handlers`: That's who you'll handle the traffic from these intercepts.
+
+Consider this specification file:
+
+```yaml
+connection:
+  namespace: telepresence-compose-demo # Consider your local setup moved to this namespace.
+workloads:
+  - name: userapi
+    intercepts:
+      - handler: userapi # Matches the handler's name.
+        global: true
+        previewURL: false
+        port: 80 # The corresponding Kubernetes service uses the port 80.
+        localPort: 8080 # The local container uses port 8080.
+handlers:
+  - name: userapi
+    docker:
+      build:
+        context: ./
+        args: # Build the docker image on the fly.
+          - --file
+          - ./docker/userapi/Dockerfile
+```
+
+Setting the namespace in the connection field, means that any container you'll define in this file, will be
+able to communicate with the cluster as if it was actually located there.
+
+We indicate that the `userapi` workload needs to be intercepted. It means that any request going to this service
+will be redirected to your local container (we call that a global intercept, but you could also do it for a subset of the
+traffic using a personal intercepts).
+
+Then we define the handler that will receive the traffic. In this case, we use `userapi`'s Dockerfile.
+
+Now let's run that spec:
+
+```cli
+> telepresence intercept run telepresence.spec.yaml
+   Intercept name    : userapi
+   State             : ACTIVE
+   Workload kind     : Deployment
+   Destination       : 127.0.0.1:8080
+   Volume Mount Point: /var/folders/x_/4x_4pfvx2j3_94f36x551g140000gp/T/telfs-3467079569
+   Intercepting      : all TCP requests
+   Layer 5 Hostname  : userapi.telepresence-compose-demo.svc.cluster.local
+Intercept spec "telepresence.spec" started successfully, use ctrl-c to cancel.
+Attempt 1: Checking PostgreSQL connection...
+PostgreSQL is ready!
+2023/11/01 20:11:24 Listening on 8080
+```
+
+If you hit the real endpoint, you will see the traffic in the logs. You can get access it
```

Comment:
Which logs, the pod logs or local logs? 🙂 we should be as specific as we can so users know exactly where to look. It can feel silly writing that way since it seems obvious but its integral in posts like these where someone might know absolutely nothing about Kubernetes, or they may be an expert, either way we need to be super specific